syncd/syncd.cpp

```diff
@@ -2382,7 +2396,56 @@ void processPfcWdEvent(
                 sai_deserialize_queue_stat(str, stat);
                 queueCounterIds.push_back(stat);
             }
-            PfcWatchdog::setQueueCounterList(queueVid, queueId, queueCounterIds);
+            PfcWatchdog::setQueueCounterList(vid, rid, queueCounterIds);
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Object tyoe for not supported");
+        }
+    }
+}
+
+void processPfcWdPluginEvent(
+        _In_ swss::ConsumerStateTable &consumer)
+{
+    std::lock_guard<std::mutex> lock(g_mutex);
+
+    SWSS_LOG_ENTER();
+
+    swss::KeyOpFieldsValuesTuple kco;
+    consumer.pop(kco);
+
+    const auto &key = kfvKey(kco);
+    const auto &op = kfvOp(kco);
+
+    if (op == DEL_COMMAND)
+    {
+        PfcWatchdog::removeCounterPlugin(key);
+        return;
+    }
+
+    const auto values = kfvFieldsValues(kco);
+    for (const auto& valuePair : values)
+    {
+        const auto field = fvField(valuePair);
+        const auto value = fvValue(valuePair);
+
+        if (field != SAI_OBJECT_TYPE)
```

Comment:
you do not need to reuse SAI_OBJECT_TYPE, this counter plugin has nothing to do with SAI. You can define your own field and value, like "type:queue", "type:port"

Reply:
For now, it passes SAI OIDs only based on the type of object. If we'll need to extend it, I'll define my own type, but at least for this purpose, it looks legitimate.